ip.go

```diff
@@ -248,6 +248,8 @@ func ExtractIPFromXFFHeader(options ...TrustOption) IPExtractor {
 		}
 		ips := append(strings.Split(strings.Join(xffs, ","), ","), directIP)
 		for i := len(ips) - 1; i >= 0; i-- {
+			ips[i] = strings.ReplaceAll(ips[i], "[", "")
```

Comment:
maybe instead of replaceAlling in for loop we can do replacement before loop or  instead of ReplaceAll we cut first/last characters off if ip starts with `[`  (something similar to comment above)

NB: this change is not covered by tests